lib/internal/fs/cp/cp-sync.js

```diff
@@ -0,0 +1,349 @@
+'use strict';
+
+// This file is a modified version of the fs-extra's copySync method
+// modified in the following ways:
+//
+// - Use of the graceful-fs has been replaced with fs.
+// - Helpers from stat.js have been inlined.
+// - Formatting and style changes to match Node.js' linting rules.
+// - Parameter validation has been added.
+
+const { areIdentical, isSrcSubdir } = require('internal/fs/cp/cp');
+const { codes } = require('internal/errors');
+const {
+  os: {
+    errno: {
+      EEXIST,
+      EISDIR,
+      EINVAL,
+      ENOTDIR,
+    }
+  }
+} = internalBinding('constants');
+const {
+  ERR_FS_CP_DIR_TO_NON_DIR,
+  ERR_FS_CP_EEXIST,
+  ERR_FS_CP_FIFO_PIPE,
+  ERR_FS_CP_NON_DIR_TO_DIR,
+  ERR_FS_CP_SOCKET,
+  ERR_FS_CP_SYMLINK_TO_SUBDIRECTORY,
+  ERR_FS_CP_TO_SUBDIRECTORY,
+  ERR_FS_CP_UNKNOWN,
+  ERR_FS_EISDIR,
+  ERR_INVALID_RETURN_VALUE,
+} = codes;
+const fs = require('fs');
+const {
+  chmodSync,
+  copyFileSync,
+  existsSync,
+  lstatSync,
+  mkdirSync,
+  readdirSync,
+  readlinkSync,
+  statSync,
+  symlinkSync,
+  unlinkSync,
+  utimesSync,
+} = fs;
+const path = require('path');
+const {
+  dirname,
+  isAbsolute,
+  join,
+  parse,
+  resolve,
+} = path;
+const { isPromise } = require('util/types');
+
+function cpSyncFn(src, dest, opts) {
+  // Warn about using preserveTimestamps on 32-bit node
+  if (opts.preserveTimestamps && process.arch === 'ia32') {
+    const warning = 'Using the preserveTimestamps option in 32-bit ' +
+      'node is not recommended';
+    process.emitWarning(warning, 'TimestampPrecisionWarning');
+  }
+  const { srcStat, destStat } = checkPathsSync(src, dest, opts);
+  checkParentPathsSync(src, srcStat, dest, 'copy');
+  return handleFilterAndCopy(destStat, src, dest, opts);
+}
+
+function checkPathsSync(src, dest, opts) {
+  const { srcStat, destStat } = getStatsSync(src, dest, opts);
+
+  if (destStat) {
+    if (areIdentical(srcStat, destStat)) {
+      throw new ERR_FS_CP_TO_SUBDIRECTORY({
+        code: 'EINVAL',
```

Comment:
Are there existing cases where we throw and document an error that starts with `ERR_` but the `code` property is different? I feel like it's going to confuse users. /cc @jasnell

Reply:
I've updated the `code` to correspond to the name of the error.

Reply:
Thanks, though I think the code is set this way automatically if you don't specify it?

Reply:
I believe you're correct:

https://github.com/nodejs/node/pull/39372/commits/fc26ded98316f3f2c3973712951adfdcc15a5710

Updated accordingly.